From: Matthias Klose <doko@debian.org>
Date: Sat, 29 Oct 2022 21:23:32 -0400
Subject: Fix regression byte-compiling filenames from stdin

Bug-upstream: https://bugs.python.org/issue45428
Forwarded: https://github.com/python/cpython/pull/28848
---
 Lib/py_compile.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Lib/py_compile.py b/Lib/py_compile.py
index 388614e..6e72299 100644
--- a/Lib/py_compile.py
+++ b/Lib/py_compile.py
@@ -194,6 +194,7 @@ def main():
     else:
         filenames = args.filenames
     for filename in filenames:
+        filename = filename.rstrip('\n')
         try:
             compile(filename, doraise=True)
         except PyCompileError as error:
